### DISCUSSION TITLE
Extremely slow python autocomplete performance

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#11467] [Troubleshooting]
Hello, 
I recently switched from Neovim to helix, and while i'm really enjoying the intuitive motions and out of the box perfomance, i'm having extremely poor performance in my autocomplete with pyright as my LSP (I also tried pylsp, and still the same problem).
I tried setting idle-timeout to 0 and completion-trigger-len = 1 like i've read here, but to no avail.
Hope you can provide some guidance. 

Thank you for your help 😄 

**Version** : helix 24.3 (2cadec0b)

**config.toml :** 

```
theme = "everforest_dark"
[editor]
idle-timeout = 0
popup-border = "all"
color-modes = true
mouse = true 
completion-trigger-len = 1


[editor.indent-guides]
render = true
character =  "▏"
skip-levels = 2
  
[editor.statusline]
left = ["mode", "spinner"]
center = ["file-name"]
right = ["version-control", "diagnostics", "selections", "position", "file-encoding", "file-line-ending", "file-type"]
separator = "│"
mode.normal = "NORMAL"
mode.insert = "INSERT"
mode.select = "SELECT"

[keys.normal]
end = "goto_file_end" 
S-d = "kill_to_line_end"
tab = "insert_tab"

```
**languages.toml :** 
```
[[language]]
name = "python"
language-servers = ["pyright"]
formatter = { command = "black", args = ["--quiet", "-"] }
auto-format = true


[language-server.pyright.config]
miccabe = {enabled = false}
pyls_isort = {enabled = false}
pycodestyle = {enabled = false, maxLineLength = 79}
pydocstyle = {enabled = false}
autopep8 = {enabled = false}
yapf = {enabled = false}
flake8 = {enabled = false}
pylint = {enabled = true}
ruff = {enabled = true, extendSelect = ["I"]}
mypy = {enabled = true, live_mode = true, strict = true}



[language-server.gopls]
command = "gopls"
config = { "gofumpt" = true, "local" = "goimports", "semanticTokens" = true, "staticcheck" = true, "verboseOutput" = true, "analyses" = { "fieldalignment" = true, "nilness" = true, unusedparams = true, unusedwrite = true, useany = true }, usePlaceholders = true, completeUnimported = true, hints = { "assignVariableType" = true, "compositeLiteralFields" = true, "compositeLiteralTypes" = true, "constantValues" = true, "functionTypeParameters" = true, "parameterNames" = true, "rangeVariableTypes" = true } }

[[language]]
name = "go"
roots = ["go.work", "go.mod"]
auto-format = true
comment-token = "//"
language-servers = [ "gopls"]

```
[helix.log](https://github.com/user-attachments/files/16635592/helix.log)

**Comment by TornaxO7** [comment]
May I ask if you could create a little video?
Have you also tried [pylyzer](https://github.com/mtshiba/pylyzer)?
May I ask why you didn't use the issue-template because some information is missing like which helix version you're using and maybe the logs.

**Reply by ghost** [reply]
Yes sorry, I added the log file

**Comment by ghost** [comment]
Ah sorry didn't see it. I will provide the version now, and try to upload a video soon. As for pylyzer, it's not as supported and feature rich as pyright or pylsp (from what've ive seen).

Do you have some leads on what could cause the problem?

Thanks for your time.

**Comment by ghost** [comment]
Also, ive the same problem with Go in gopls. Maybe that can connect some dots?

**Reply by leo-unglaub** [reply]
I have the same problem with PHP (phpactor)

**Reply by leo-unglaub** [reply]
Here is my LSP log in such a case:
[helix-lsp.log](https://github.com/user-attachments/files/16606686/helix-lsp.log)

**Comment by kirawi** [comment]
Can you try `24.7`?

**Reply by ghost** [reply]
will try and keep you updated! thanks

**Reply by ghost** [reply]
No improvements in 24.7

**Comment by leo-unglaub** [comment]
Could it be, that this is related to https://github.com/helix-editor/helix/discussions/11468?
